x/mint/keeper/keeper_test.go

```diff
@@ -719,3 +719,348 @@ func (suite *KeeperTestSuite) TestDistributeDeveloperRewards() {
+		},
+		"valid case: 100.6 - 100 + 0.4 = 1; all distributed and none persisted": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			preExistingDelta:  sdk.NewDecWithPrec(4, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.OneInt(),
+			expectedPersistedInStore:  sdk.ZeroDec(),
+		},
+		"valid case: 100.8 - 100 + 0.4 = 1.2; 1 distributed, 0.2 persisted in store; mint module account": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(4, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1008, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.OneInt(),
+			expectedPersistedInStore:  sdk.NewDecWithPrec(2, 1),
+		},
+		"valid case: 100 - 100 + 0 = 0; none distributed or persisted in store; mint module account": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.ZeroDec(),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDec(100)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.ZeroInt(),
+			expectedPersistedInStore:  sdk.ZeroDec(),
+		},
+		"invalid module account name - error": {
+			moduleAccountName: poolincentivestypes.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(2, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectErr: true,
+		},
+		"provisions are less than amount distributed - error": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(2, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(101),
+
+			expectErr: true,
+		},
+		"attempted to distribute more than developer vesting module account balance - error": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			preExistingDelta:  sdk.NewDecFromInt(sdk.NewInt(keeper.DeveloperVestingAmount)).Add(sdk.OneDec()),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectErr: true,
+		},
+	}
+	for name, tc := range tests {
+		suite.Run(name, func() {
+			// Setup.
+			suite.Setup()
+			mintKeeper := suite.App.MintKeeper
+			ctx := suite.Ctx
+
+			if !tc.preExistingDelta.IsNil() {
+				mintKeeper.SetTruncationDelta(ctx, tc.moduleAccountName, tc.preExistingDelta)
+			}
+
+			// System under test
+			actualAmount, err := mintKeeper.HandleTruncationDelta(ctx, tc.moduleAccountName, tc.provisions, tc.amountDistributed)
+
+			// Assertions.
+			if tc.expectErr {
+				suite.Require().Error(err)
+				return
+			}
+
+			suite.Require().NoError(err)
+			suite.Require().Equal(tc.expectedDistributedAmount.String(), actualAmount.String())
```

Comment:
Since handletruncatedelta handles more logic than calculatetruncate delta, I think it would be better if we test extra logics that are happening in handle truncate delta. Would it make sense to have extra cases for community pool funding or module set gauge?